rules-tests/Php55/Rector/String_/StringClassNameToClassConstantRector/Fixture/no_pre_slash.php.inc

```diff
@@ -16,11 +16,11 @@ final class PreSlash
 
 namespace Rector\Tests\Php55\Rector\String_\StringClassNameToClassConstantRector\Fixture;
 
-final class PreSlash
+final class NoPreSlash
 {
     public function preSlash()
     {
-        return '\\' . \Rector\Tests\Php55\Rector\String_\StringClassNameToClassConstantRector\Source\AnotherClass::class;
+        return \Rector\Tests\Php55\Rector\String_\StringClassNameToClassConstantRector\Source\AnotherClass::class;
```

Comment:
This is invalid, the result is different, the comparison printed  first has pre-backslash, the later it don't have as using ::class

Reply:
See why it is invalid

https://3v4l.org/I947K

Reply:
Yes, it's different, but in practise this value is prefered. The class must exist and does in both cases.

Reply:
The CI4 code should be update then to use the class name, not require a slash.  Seems like a hack :)
 
In the Mautic case it added slash that made code less readable.

Reply:
It is a special rule for CI4 routing. It distinguishes between `Foo` (relative to the default namespace) and `\Foo` (FQCN).
I know it is not common usage, but we cannot to change. If we change, the existing apps will break.

Reply:
I think that's rather edge case compared to common use of `::class`, so we'll stick with behavior above as default.

Reply:
@TomasVotruba how about make it configurable, eg, add config flag `keepBackslash` ?

Reply:
This:

> It distinguishes between Foo (relative to the default namespace) and \Foo (FQCN).

They are not the same. For instance, you could have classes `SomeNamespace\StringUtils` and `StringUtils`. Then, referencing `StringUtils` or `\StringUtils` from `SomeNamespace\ObjectUtils` will be referencing 2 different classes.

So I think that you'd rather keep that pre-backlash if you're targeting the class on the root, or it may produce bugs.

Reply:
I agree with leoloso.

maybe we can detect the mautic case somehow instead of removing the slash in the general case?

Reply:
@staabm I had similar issues with othe projects that we had to remove manually.

Could you share link to piece of code where this is relevant?

Reply:
@leoloso 

> They are not the same. For instance, you could have classes SomeNamespace\StringUtils and StringUtils. Then, referencing StringUtils or \StringUtils from SomeNamespace\ObjectUtils will be referencing 2 different classes.

This is different use case. I talk about concatted slash string:

```diff
+'\\' . SomeClass::class
-SomeClass::class
```

Two same named classes are kept separated.

Reply:
Probably the following test will break with this PR.
https://github.com/codeigniter4/CodeIgniter4/pull/8321/commits/7646e3fc97a9c99fb12133eb29f8a72832642d02

Reply:
@kenjis yes, that will break that

Reply:
I create new PR to make it configurable:

- https://github.com/rectorphp/rector-src/pull/5354